robot_smach_states/src/robot_smach_states/navigation/navigate_to_grasp.py

```diff
@@ -52,14 +52,14 @@ def generateConstraint(self):
 
         try:
             pose = entity.pose #TODO Janno: Not all entities have pose information
-            x = pose.position.x
-            y = pose.position.y
+            x = pose.frame.p.x()
+            y = pose.frame.p.y()
         except KeyError, ke:
             rospy.logerr("Could not determine pose: ".format(ke))
             return None
 
         try:
-            rz = entity.pose.orientation.z
+            rz, _, _ = entity.pose.frame.M.GetEulerZYX()
```

Comment:
If there are also nonzero roll and pitch angles, this will generally not return the correct projected yaw angle as the other angles may also influence this. I guess this is out of the scope of this PR, as this problem already existed in the old version, but an issue should be filed for this.